### PR TITLE
[Feature] support show data and show tablet for MV

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Support `show data` and `show tablet` for materialized-view.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java [hunk 1]
```diff
@@ -1195,7 +1195,7 @@ private void handleShowTablet() throws AnalysisException {
                 if (table == null) {
                     ErrorReport.reportAnalysisException(ErrorCode.ERR_BAD_TABLE_ERROR, showStmt.getTableName());
                 }
-                if (!(table instanceof OlapTable)) {
+                if (!table.isOlapTable() && !table.isMaterializedView()) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:1198]: isLocalTable

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeChecker.java [hunk 2]
```diff
@@ -1043,7 +1043,7 @@ public int compare(Table t1, Table t2) {
                         ErrorReport.reportAnalysisException(ErrorCode.ERR_BAD_TABLE_ERROR, tableName);
                     }
 
-                    if (table.getType() != Table.TableType.OLAP) {
+                    if (!table.isOlapTable() && !table.isMaterializedView()) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeChecker.java:1046]: isLocalTable

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|1|100.00%|[]|

**Review by kangkaisen — DISMISSED** [review]

**Review by ABingHuang — DISMISSED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|1|100.00%|[]|

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**